sway-core/src/decl_engine/functional_decl_id.rs

```diff
@@ -10,6 +10,7 @@ use crate::{
 pub enum FunctionalDeclId {
     TraitFn(DeclId<ty::TyTraitFn>),
     Function(DeclId<ty::TyFunctionDeclaration>),
+    Constant(DeclId<ty::TyConstantDeclaration>),
```

Comment:
As discussed before, the name of this `enum` is now a bit confusing. You may want to create an issue to decide on a new name.